src/test/java/edu/cornell/kfs/coa/document/validation/impl/GlobalIndirectCostRecoveryAccountsRuleTest.java

```diff
@@ -0,0 +1,281 @@
+package edu.cornell.kfs.coa.document.validation.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.kuali.kfs.coa.businessobject.AccountGlobalDetail;
+import org.kuali.kfs.coa.businessobject.IndirectCostRecoveryAccount;
+
+import edu.cornell.kfs.coa.businessobject.CuAccountGlobal;
+import edu.cornell.kfs.coa.businessobject.IndirectCostRecoveryAccountChange;
+import edu.cornell.kfs.coa.businessobject.SubAccountGlobal;
+import edu.cornell.kfs.coa.businessobject.SubAccountGlobalDetail;
+import edu.cornell.kfs.coa.fixture.AccountGlobalDetailFixture;
+import edu.cornell.kfs.coa.fixture.AccountGlobalFixture;
+import edu.cornell.kfs.coa.fixture.SubAccountGlobalDetailFixture;
+import edu.cornell.kfs.coa.fixture.SubAccountGlobalFixture;
+
+public class GlobalIndirectCostRecoveryAccountsRuleTest {
+
+	private GlobalIndirectCostRecoveryAccountsRule globalIndirectCostRecoveryAccountsRule;
+
+	private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger
+			.getLogger(GlobalIndirectCostRecoveryAccountsRuleTest.class);
+
+	@Before
+	public void setUp() {
+		globalIndirectCostRecoveryAccountsRule = new GlobalIndirectCostRecoveryAccountsRule();
+	}
+
+	@Test
+	public void testAccountGlobalCheckICRAccuntTotalDistributionOnDetailWillBe100PercentAfterUpdate_Pass() {
+		LOG.debug("enter testAccountGlobalCheckICRAccuntTotalDistributionOnDetailWillBe100PercentAfterUpdate_Pass");
+		CuAccountGlobal accountGlobal = (CuAccountGlobal) AccountGlobalFixture.ACCT_GLOBAL_3333333_100_ACTIVE_1111111_2222222_98_2_INACTIVE
+				.getAccountGlobal();
+		AccountGlobalDetail accountGlobalDetail = AccountGlobalDetailFixture.ACCOUNT_GLOBAL_DETAIL_1111111_2222222_98_2
+				.getAccountGlobalDetail();
+
+		LOG.debug("updates");
+		List<IndirectCostRecoveryAccountChange> updates = accountGlobal
+				.getIndirectCostRecoveryAccounts();
+		for (IndirectCostRecoveryAccountChange icrChange : updates) {
+			LOG.debug("Chart " + icrChange.getIndirectCostRecoveryFinCoaCode()
+					+ " account "
+					+ icrChange.getIndirectCostRecoveryAccountNumber()
+					+ " percent " + icrChange.getAccountLinePercent()
+					+ " active " + icrChange.isActive());
+		}
```

Comment:
There seems to be a lot of duplication in these test methods, especially for these debugging loops. I would recommend creating some new methods to consolidate the repeating parts where it makes sense.